docs/reconcilation.md

```diff
@@ -0,0 +1,72 @@
+# Kubernetes reconciliation
+
+In this document we will describe the basics of Kubernetes reconciliation and specifically in the context of the [FoundationDB operator](https://github.com/FoundationDB/fdb-kubernetes-operator).
+
+## Reconciliation
+
+Kubernetes has a declarative way to describe the desired state of resources.
+Each resource is watched/controlled by a so called controller or operator.
+If a resource gets created or updated the contoller will compare the desired state against the current state (reconcile).
+If there is any difference the controller will take action to reach the desired state e.g. by creating/deleting Pods.
+Even though the controller normally reacts based on watches (the Kubernetes API will send updates if a resource is changed) the controller will always compare the complete state.
+This is also known as [level triggered](#further_reading) and has the benefit that we don't care to much about a missed event (which can easily happen in a distributed system).
+Each controller can define for which resources it wants to be notified, normally this includes all resources that the controller "own" e.g. manages.
+Besides the active watch a controller also periodically sync it's state (per default the synchronization period is `10h`) that means after the period is over the controller will compare the desired state with the current state for all resources managed by the controller.
+
+## FoundationDB operator
+
+The same pattern applies to the FoundationDB operator with a `FoundationDBCluster` manifest we describe the desired state of a FoundationDB cluster running in Kubernetes.
+The FDB operator implements [multiple sub-reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.33.0/controllers/cluster_controller.go#L123-L149) mostly for a better readability and to separate the individual tasks.
+As an example there is a sub-reconciler that only handles the ConfigMap reconcilation and there is another sub-reconciler that deletes Pod.
+Why should I care about these sub-reconcilers?
+Each sub-reconciler will be called serially in the defined order.
+During a reconciliation "step" the sub-reconciler can decide if the reconciliation can continue or not.
+That means that a sub-reconciler can block the reconciliation from completion.
+
+### How to debug a stuck reconciliation
+
+A reconciliation can get stuck if the desired state can't be reached e.g. because an upstream resource won't be available (Pod is stuck in `Pending`).
+There are two places in the `FoundationDBCluster` status that can provide further guidance on why the reconciliation is stuck:
+
+- The [generations](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.33.0/api/v1beta1/foundationdbcluster_types.go#L670) that describe the latest reconciled generation (if present) and any additional generations (and the reason) that are not fully reconciled. Most of the conditions are fairly well describing.
```

Comment:
'fairly well describing' doesn't make sense to me - do you perhaps mean 'fairly well described' or 'self describing'?